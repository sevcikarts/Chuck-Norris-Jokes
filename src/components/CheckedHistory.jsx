import React from "react";
import Switch from "@material-ui/core/Switch";
import FormControlLabel from "@material-ui/core/FormControlLabel";

const CheckedHistory = ({ checkedHistry, setCheckedHistry,listOfJokes }) => {
  function scrollWin() {
    setTimeout(() => {
      window.scrollTo({
        top: document.body.scrollHeight,
        behavior: "smooth",
      });
    }, 500);
  }
  console.log(document.body.scrollHeight);

  return (
    <div>
      <FormControlLabel
        control={
          <Switch
            checked={checkedHistry}
            onChange={(e) => setCheckedHistry(e.target.checked)}
            onClick={() => scrollWin()}
            color="primary"
          />
        }
        label= { `show searched jokes (${listOfJokes.length-1})` }
      />
    </div>
  );
};

export default CheckedHistory;
