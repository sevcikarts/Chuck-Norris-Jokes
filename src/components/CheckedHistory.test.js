import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16"
import toJSON from "enzyme-to-json"
import Switch from "@material-ui/core/Switch";
import FormControlLabel from "@material-ui/core/FormControlLabel";
import CheckedHistory from "./CheckedHistory.jsx";

Enzyme.configure({ adapter: new Adapter() });

it("rendering TextField", () => {
    const wrapper = shallow(<CheckedHistory  />);
    const input = wrapper.find(<FormControlLabel list ={
        listOfJokes.length-1
    }></FormControlLabel>);
    expect(input.length).toBe(1);
  });

  it("snapShot", () => {
    const wrapper = shallow(
      <FormControlLabel />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });