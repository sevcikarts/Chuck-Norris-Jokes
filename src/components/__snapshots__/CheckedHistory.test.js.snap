// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapShot 1`] = `
<ForwardRef(FormControlLabel)
  classes={
    Object {
      "disabled": "Mui-disabled",
      "label": "MuiFormControlLabel-label",
      "labelPlacementBottom": "MuiFormControlLabel-labelPlacementBottom",
      "labelPlacementStart": "MuiFormControlLabel-labelPlacementStart",
      "labelPlacementTop": "MuiFormControlLabel-labelPlacementTop",
      "root": "MuiFormControlLabel-root",
    }
  }
/>
`;
